python/private/whl_target_platforms.bzl

```
@@ -118,68 +103,129 @@ def _whl_priority(value):
     # Windows does not have multiple wheels for the same target platform
     return (False, False, 0, 0)
 
-def select_whl(*, whls, want_abis, want_platform):
-    """Select a suitable wheel from a list.
+def select_whls(*, whls, want_version = "3.0", want_abis = [], want_platforms = []):
+    """Select a subset of wheels suitable for target platforms from a list.
 
     Args:
         whls(list[struct]): A list of candidates.
+        want_version(str): An optional parameter to filter whls by version. Defaults to '3.0'.
         want_abis(list[str]): A list of ABIs that are supported.
-        want_platform(str): The target platform.
+        want_platforms(str): The platforms
 
     Returns:
         None or a struct with `url`, `sha256` and `filename` attributes for the
         selected whl. If no match is found, None is returned.
     """
     if not whls:
-        return None
+        return []
+
+    version_limit = -1
+    if want_version:
+        version_limit = int(want_version.split(".")[1])
 
     candidates = {}
     for whl in whls:
         parsed = parse_whl_name(whl.filename)
-        if parsed.abi_tag not in want_abis:
-            # Filter out incompatible ABIs
-            continue
 
-        platform_tags = list({_LEGACY_ALIASES.get(p, p): True for p in parsed.platform_tag.split(".")})
+        supported_implementations = {}
```

Ah, is this dict acting like a set? Just drop a comment. I really wish starlark had a set type
```suggestion
        # dict, but used as a set.
        supported_implementations = {}
```